src/backup.ts

```diff
@@ -0,0 +1,478 @@
+  instanceFormattedName_: string;
+  request: <T, R = void>(
+    config: RequestConfig,
+    callback: RequestCallback<T, R>
+  ) => void;
+  constructor(instance: Instance, name: string) {
+    this.request = instance.request;
+    this.instanceFormattedName_ = instance.formattedName_;
+    this.formattedName_ = Backup.formatName_(instance.formattedName_, name);
+    this.id = this.formattedName_.split('/').pop() || '';
+  }
+
+  create(options: CreateBackupOptions): Promise<CreateBackupResponse>;
+  create(options: CreateBackupOptions, callback: CreateBackupCallback): void;
+  /**
+   * @typedef {array} CreateBackupResponse
+   * @property {Backup} 0 The new {@link Backup}.
+   * @property {Operation} 1 An {@link Operation} object that can be used to check
+   *     the status of the request.
+   * @property {object} 2 The full API response.
+   */
+  /**
+   * @callback CreateBackupCallback
+   * @param {?Error} err Request error, if any.
+   * @param {Backup} backup The new {@link Backup}.
+   * @param {Operation} operation An {@link Operation} object that can be used to
+   *     check the status of the request.
+   * @param {object} apiResponse The full API response.
+   */
+  /**
+   * Create a backup.
+   *
+   * @method Backup#create
+   * @param {CreateBackupOptions} options Configuration object.
+   * @param {CreateBackupCallback} [callback] Callback function.
+   * @returns {Promise<CreateBackupResponse>} when resolved, the backup
+   *     operation will have started, but will not have necessarily completed.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const oneDay = 1000 * 60 * 60 * 24;
+   * const expiryTime = new PreciseDate(Date.now() + oneDay);
+   * const backup = instance.backup('my-backup');
+   * const [, backupOperation] = await backup.create(
+   *   'projects/my-project/instances/my-instance/databases/my-database',
+   *   expiryTime
+   * );
+   * // Await completion of the backup operation.
+   * await backupOperation.promise();
+   */
+  create(
+    options: CreateBackupOptions,
+    callback?: CreateBackupCallback
+  ): Promise<CreateBackupResponse> | void {
+    const gaxOpts: CallOptions = options.gaxOptions as CallOptions;
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.ICreateBackupRequest = {
+      parent: this.instanceFormattedName_,
+      backupId: this.id,
+      backup: {
+        database: options.databasePath,
+        expireTime: options.expireTime.toStruct(),
+        name: this.formattedName_,
+      },
+    };
+    return this.request(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'createBackup',
+        reqOpts,
+        gaxOpts,
+      },
+      (err, resp) => {
+        if (err) {
+          callback!(err, null, null, resp!);
+          return;
+        }
+        callback!(null, this, resp, resp);
+      }
```

Comment:
Hmmm I don't think these callback args are correct. I tried to make it `(err, operation, resp)` but the compiler keeps telling me that `operation` is undefined. Maybe this is what @AVaksman was referring to in the comment above. I don't quite know how to fix this though.